### ISSUE TITLE
[clang-tidy] False positive for `cppcoreguidelines-init-variables`

### ISSUE THREAD

**Issue by timxx** [llvm/llvm-project#173435]
Given any for loop objc code, it reports variable is not initialized:

```objc
for (NSNumber* n in array) {
}
```

here `array` is `NSArray<NSNumber*>>`.

It can be reproduce with other types, such as [allPresses](https://developer.apple.com/documentation/uikit/uipressesevent/allpresses?language=objc) in [UIPressesEvent](https://developer.apple.com/documentation/uikit/uipressesevent?language=objc)

```objc
for (UIPress* press in event.allPresses) {
}
```
